### PR TITLE
B-04004 simplify

### PR DESCRIPTION
**Pull request by robbiecarlton**
Goes with this version of elemental-chat dna - https://github.com/holochain/elemental-chat/pull/48

Big refactor, simplifying some things, moving some things around. 

Big changes: 

Removed use of indexedb to locally cache data. Local storage is now just used for some stats calculation, the user handle and most recent channel.

Reorganized store, separating it into
`index`: for some basic UI stuff
`holochain`: storing the hc and holo clients and associated data
`elementalChat`: storing elemental chat specific data (channels and messages)

Updated the way signaling works to use `signal_specific_chatters` with the `include_active_chatters` option.

Added some simple unit tests for the store to catch message merge issues.

As well as lot's of smaller changes.

### CODE REVIEW

**Comment by zippy** [comment]
Looks great.  Please remember to merge in the build-script changes too!

**Review by JettTech — CHANGES_REQUESTED** [review]
Overall, looks good.  I did notice that the `updateHandle` functionality isn't working, so let's get that fixed before the merge.  Otherwise, I just left a few non-blocking requests/comments.

File: package.json [hunk 1]
```diff
@@ -3,43 +3,53 @@
   "version": "0.0.1-alpha29",
   "private": true,
   "scripts": {
+    "serve:holo": "VUE_APP_UI_VERSION=$npm_package_version VUE_APP_CONTEXT='holo-host' VUE_APP_CHAPERONE_SERVER_URL='http://localhost:24273/' vue-cli-service serve",
+    "build": "VUE_APP_UI_VERSION=$npm_package_version VUE_APP_CONTEXT='holo-host' VUE_APP_CHAPERONE_SERVER_URL='https://chaperone.holo.host/' vue-cli-service build",
+    "test:unit": "vue-cli-service test:unit",
+    "lint": "vue-cli-service lint",
+    "build-self-hosted": "VUE_APP_UI_VERSION=$npm_package_version VUE_APP_CONTEXT='self-hosted' vue-cli-service build",
+    "hc:agent-1": "holochain-run-dna -p 8888 -i elemental-chat-1 -u kitsune-proxy://CIW6PxKxsPPlcuvUCbMcKwUpaMSmB7kLD8xyyj4mqcw/kitsune-quic/h/165.22.32.11/p/5778/-- ../elemental-chat/elemental-chat.dna.gz",
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply package.json:11]: good call

File: src/ElementalChat.vue [hunk 2]
```diff
@@ -179,63 +136,57 @@
...
-      showAdd: false,
-      refreshKey: 0
+      showingStats: false
     }
   },
   methods: {
+    ...mapActions(['editHandle']),
     ...mapActions('elementalChat', [
       'listChannels',
-      'updateHandle',
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply src/ElementalChat.vue:198]: No, just didn't need the separate "updateHandle" action as we already have `setAgentHandle` mutation

File: src/store/__tests__/elementalChat.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import store from '../index'
+
+jest.mock('../callZome')
+
+const makeChannel = (uuid, name) => ({
+  info: {
+    name,
+    created_by: ''
+  },
+  entry: { category: 'General', uuid },
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply src/store/__tests__/elementalChat.test.js:12]: I think that would be fine, but I don't think it matters either way so will just leave it as is, unless you feel strongly

File: src/store/callZome.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    fn_name: fnName,
+    provenance: state.agentKey,
+    payload
+  },
+  timeout)
+
+const LOG_ZOME_CALLS = true
+
+export const callZome = async (dispatch, rootState, zomeName, fnName, payload, timeout) => {
+  if (LOG_ZOME_CALLS) {
```
[the remark this answers is not in the document]
Comment (robbiecarlton) [reply src/store/callZome.js:25]: Yup can do

**Review by zippy — APPROVED** [review]
Woot!

**Pull request merged**